SampleProducer/SampleProducer.cs

```diff
@@ -60,11 +60,17 @@ public static void Main(string[] args)
                 var CreateStreamResponse = kinesisClient.CreateStreamAsync(createStreamReq).Result;
                 Console.Error.WriteLine("Created Stream : " + myStreamName);
             }
-            catch (ResourceInUseException)
+            catch (Exception ex)
             {
-                Console.Error.WriteLine("Producer is quitting without creating stream " + myStreamName +
-                    " to put records into as a stream of the same name already exists.");
-                Environment.Exit(1);
+                if (ex is AggregateException || ex is ResourceInUseException)

```

Comment:
Under what circumstances will `AggregateException` happen?

Reply:
`AggregateException` is the parent exception of `ResourceInUseException`, yeah i think another level of if: `ex is AggregateException && ex is ResourceInUseException` is more appropriate!

Reply:
Hey, improved the error handling!